clang-tools-extra/clangd/unittests/HoverTests.cpp

```diff
@@ -3941,6 +3945,157 @@ TEST(Hover, DisableShowAKA) {
   EXPECT_EQ(H->Type, HoverInfo::PrintedType("m_int"));
 }
 
+TEST(Hover, ShowFields) {
+  struct {
+    const char *const Code;
+    const std::function<void(HoverInfo &)> ExpectedBuilder;
+  } Cases[] = {
+    {
+      R"cpp(// Struct
+        namespace ns1 {
+          struct MyClass {
+            // Public fields shown in hover
+            int field1;
+            int field2;
+
+            // Methods and private fields not shown
+            void method();
```

Comment:
would it make sense to add an additional option that also shows public methods? (probably out-of-scope for this PR though)